src/vm/dllimport.cpp

```diff
@@ -6034,62 +6036,76 @@ HMODULE NDirect::LoadFromNativeDllSearchDirectories(AppDomain* pDomain, LPCWSTR
 }
 
 #ifdef FEATURE_PAL
-static void DetermineLibNameVariations(const WCHAR* const** libNameVariations, int* numberOfVariations, const SString& libName, bool libNameIsRelativePath)
+static void DetermineLibNameVariations(const WCHAR** libNameVariations, int* numberOfVariations, const SString& libName, bool libNameIsRelativePath)
 {
-    if (libNameIsRelativePath)
+    // Supported lib name variations
+    static const int MaxFmtCount = 4;
+    static auto NameFmt = W("%.0s%s%.0s");
+    static auto PrefixNameFmt = W("%s%s%.0s");
+    static auto NameSuffixFmt = W("%.0s%s%s");
+    static auto PrefixNameSuffixFmt = W("%s%s%s");
+
+    int varCount = 0;
+    if (!libNameIsRelativePath)
+    {
+        _ASSERTE(*numberOfVariations >= 1);
+        libNameVariations[varCount++] = NameFmt;
+    }
+    else
     {
         // We check if the suffix is contained in the name, because on Linux it is common to append
         // a version number to the library name (e.g. 'libicuuc.so.57').
-        bool containsSuffix;
+        bool containsSuffix = false;
         SString::CIterator it = libName.Begin();
         if (libName.Find(it, PLATFORM_SHARED_LIB_SUFFIX_W))
         {
-            it += wcslen(PLATFORM_SHARED_LIB_SUFFIX_W);
+            it += COUNTOF(PLATFORM_SHARED_LIB_SUFFIX_W);
             containsSuffix = it == libName.End() || *it == (WCHAR)'.';
         }
-        else
-        {
-            containsSuffix = false;
-        }
 
+        // If the path contains a path delimiter, we don't add a prefix
+        it = libName.Begin();
+        bool containsDelim = libName.Find(it, PLATFORM_PATH_DELIM);
+
+        _ASSERTE(*numberOfVariations >= MaxFmtCount);
         if (containsSuffix)
         {
-            static const WCHAR* const SuffixLast[] =
-            {
-                W("%.0s%s"),   // name
-                W("%s%s%.0s"), // prefix+name
-                W("%.0s%s%s"), // name+suffix
-                W("%s%s%s")    // prefix+name+suffix
-            };
-            *libNameVariations = SuffixLast;
-            *numberOfVariations = COUNTOF(SuffixLast);
+            libNameVariations[varCount++] = NameFmt;
```

Comment:
Suggestion:
Move these two commons code outside if/else
such as 
libNameVariations[varCount++] = NameFmt;
libNameVariations[varCount++] = NameSuffixFmt;
if(containsSuffix)
{
...
}
else
{
...
}

Reply:
I can do that, but that will change the previously defined search path. Is that what we want? I was keeping the same search order since I thought it was intentional.

Reply:
oh.. my fault. I forgot that this order is our search order.